akka-cluster/src/main/scala/akka/cluster/ClusterEvent.scala

```diff
@@ -278,6 +294,24 @@ object ClusterEvent {
   /**
    * INTERNAL API
    */
+  private[cluster] def diffUnreachableTeam(oldGossip: Gossip, newGossip: Gossip, selfUniqueAddress: UniqueAddress): immutable.Seq[UnreachableDataCenter] = {
+    if (newGossip eq oldGossip) Nil
+    else {
+      val oldUnreachableNodes = oldGossip.overview.reachability.allUnreachableOrTerminated
+
+      newGossip.overview.reachability.observersGroupedByUnreachable.collect {
+        case (unreachable, observers) if !oldUnreachableNodes.contains(unreachable) &&
+          unreachable != selfUniqueAddress &&
+          newGossip.member(unreachable).dataCenter != newGossip.member(selfUniqueAddress).dataCenter &&
+          observers.exists(newGossip.member(_).dataCenter == newGossip.member(selfUniqueAddress).dataCenter) ⇒
+          UnreachableDataCenter(newGossip.member(unreachable).dataCenter)
+      }(collection.breakOut)
+    }
+  }
```

Comment:
Perhaps the above is correct, but it looks complicated.
Given 3 DCs: dc1, dc2, dc3 (that can be found from `oldGossip.allDataCenters union newGossip.allDataCenter`)

Foreach such `dc` find observations (`Reachability.Record`) from nodes `selfDc` -> to nodes in that`dc`, all other observations  should be thrown away.

Given https://github.com/akka/akka/pull/23291/files#diff-8809e33e621724853e25d776e9969c9bR67
```
 lazy val dcReachability: Reachability =
     overview.reachability.removeObservers(
       members.collect { case m if m.dataCenter != selfDc ⇒ m.uniqueAddress })
```

and adding something like this:
```
  def isReachable(otherDc: DataCenter): Boolean = {
    val reachabilityForOtherDc =
      dcReachability.remove(members.collect { case m if m.dataCenter != otherDc ⇒ m.uniqueAddress })
    reachabilityForOtherDc.isAllReachable
  }
```

Wouldn't the diff then become comparing those `isReachable(dc)` Boolean from old and new state, foreach dc.